### PR TITLE
multiple sentries per resource

### PR DESCRIPTION
**Pull request by jkminneti**
`tuber/sentryUrl` will work fine for backwards compatibility
add as many as you want with that prefix

### CODE REVIEW

File: pkg/core/releaser.go [hunk 1]
```diff
@@ -554,11 +559,11 @@ func (r releaser) goWatch(resource appResource, timeout time.Duration, errors ch
 			wg.Done()
 		}(errors, wg)
 
-		if resource.sentryUrl != "" {
+		for _, url := range resource.sentryUrls {
 			wg.Add(1)
 			go func(errors chan rolloutError, wg *sync.WaitGroup) {
 				_, logger := resource.scopes(r)
-				healthy, message := monitor.Sentry(logger, resource.sentryUrl, r.sentryBearerToken, resource.watchDuration)
+				healthy, message := monitor.Sentry(logger, url, r.sentryBearerToken, resource.watchDuration)
```
Comment (jaysonesmith) [inline pkg/core/releaser.go:566]: [C2 - Potentially broken under certain circumstances](https://production.admin.freshlyservices.net/engineering-handbook/docs/processes/software-development/code-review/#c2---potentially-broken-under-certain-circumstances)

VS Code alerted me to the use of `url` here.

Screen Shot 2021-08-09 at 12 50 46 PM

I did a google and found [this](https://eli.thegreenplace.net/2019/go-internals-capturing-loop-variables-in-closures/) page which says that due to the way go's internals work, we'd only ever add the last url in `resource.sentryUrls` `len(resource.sentryUrls)` times.  AFAIK the fix would be to pass the url into the closure like below?

```go
			go func(errors chan rolloutError, wg *sync.WaitGroup, url string) {
				_, logger := resource.scopes(r)
				healthy, message := monitor.Sentry(logger, url, r.sentryBearerToken, resource.watchDuration)
				if !healthy {
					errors <- rolloutError{
						err:                fmt.Errorf("sentry monitoring found failure"),
						resource:           resource,
						monitorFail:        true,
						monitorFailMessage: message,
					}
				}
				wg.Done()
			}(errors, wg, url)
```

**Review by jaysonesmith — CHANGES_REQUESTED** [review]
Just the one. :)

File: pkg/core/releaser.go (hunk 1 above)
Comment (jaysonesmith) [reply pkg/core/releaser.go:566]: I've verified this locally with a test function similar to the current implementation and got:

```go
urls := []string{"url-one", "url-two", "url-three"}

// out:
// i: 2, url: "url-three"
// i: 2, url: "url-three"
// i: 2, url: "url-three"
```

Comment (jkminneti) [reply pkg/core/releaser.go:566]: YUUUUUP shoot nice catch

Comment (jkminneti) [reply pkg/core/releaser.go:566]: in fact how did neither goland nor staticcheck catch that wtf

**Review by jaysonesmith — APPROVED** [review]

**Pull request merged**